test/models/tag_test.js

```diff
@@ -0,0 +1,242 @@
+import { mkdbtmp } from '../support/db'
+
+import { project as projectModel, item as itemModel, tag as tagModel } from '../../src/models'
+
+
+describe('models', async () => {
+  describe('tag', () => {
+    let db
+
+    mkdbtmp(x => db = x,
+      'db_test.sqlite', projectModel.create, { name: 'Test Project' })
+    // required to set up database schema
+    // TODO is there a better way to do this?
```

Comment:
I think this is fine, but two things come to mind: I'd be interested to explore using an in-memory db for such kind of tests, because they're almost certainly going to be faster and we won't have to clean up the file afterwards. If I'm not mistaken then you should be able to call `Database.create` with path ":memory:" to create an in-memory db. We could adjust the mkdbtmp helper to skip any file related actions if that's the path.

The second thought was that we could integrate this into the fixture helpers maybe.

```js
let p = F.project.create()
```

And this works sort of like `mkdbtmp` in that it adds before and after hooks which create the and close the db connection accessible under `p.current` (sort of like React's refs). In fact we should change `mkdbtmp` to also use that pattern then we can skip the callback and the tests are much easier to read. What do you say?